### PR TITLE
Show all apps for domain reporter on applications page

### LINKED ISSUE #833: vTHL3: Domain reporter role
A user with rights to see all applications. Can only view applications, not do any actions unless given the rights as usual.

- [x] domain reporter role
- [x] user rights added to the database and/or API (no UI yet)
- [x] domain reporter has a reports page that shows all applications (no other reports yet)

### PR DESCRIPTION
**Pull request by luontola**
closes #833

### CODE REVIEW

File: src/cljs/rems/applications.cljs [hunk 1]
```diff
@@ -3,20 +3,23 @@
...
             [rems.spinner :as spinner]
             [rems.text :refer [localize-state localize-time text]]
-            [rems.util :refer [fetch]]))
+            [rems.util :refer [fetch]]
+            [rems.roles :as roles]))
 
 (rf/reg-event-fx
  ::enter-page
  (fn [{:keys [db]} _]
    {:db (dissoc db ::my-applications)
```
Comment (hukka) [inline src/cljs/rems/applications.cljs]: Current logic dissocs :my-applications, when fetch starts. Should same be done to :all-applications?

I think that either all applications should follow the same logic, or the discrepancy should be documented.

File: src/cljs/rems/applications.cljs [hunk 2]
```diff
@@ -37,6 +40,32 @@
...
+(rf/reg-sub
+ ::all-applications
+ (fn [db _]
+   (::all-applications db)))
+
+(rf/reg-sub
+ ::loading-all-applications?
+ (fn [db _]
+   (::loading-all-applications? db)))
+
```
Comment (hukka) [inline src/cljs/rems/applications.cljs]: These are very similar to my-applications. Would it make sense to make a generic that takes an URL and a db path, and stores the data and loading status under a map in that path?

Or is that abusing re-frame or just premature abstraction, as this is just the second version and only about 20 lines?

File: src/cljs/rems/applications.cljs [hunk 3]
```diff
@@ -54,20 +83,34 @@
 
 ;;;; UI
 
+;; XXX: the application lists share sorting and filtering state
```
Comment (hukka) [inline src/cljs/rems/applications.cljs]: If we change to a model where all application data fetch state is in under one path in db, we could put the filter state there too.

File: src/cljs/rems/roles.cljs [hunk 4]
```diff
@@ -3,12 +3,14 @@
...
-  (some #{:handler :commenter :decider :past-commenter :past-decider :reporter} roles))
+(defn show-all-applications? [roles]
+  (some #{:reporter} roles))
 
-(defn is-admin? [roles]
-  ;; TODO: add admin role (owner is a business user; admin is a technical user)
+(defn show-reviews? [roles]
+  (some #{:handler :commenter :decider :past-commenter :past-decider} roles))
+
+(defn show-admin-pages? [roles]
```
Comment (hukka) [inline src/cljs/rems/roles.cljs]: Is this now rather permissions.cljs, not roles? Pretty nitpicky. I agree that `is-applicant-or-member` was not really the most descriptive name to the relevant action/permission.

File: src/cljs/rems/applications.cljs (hunk 2 above)
Comment (luontola) [reply src/cljs/rems/applications.cljs]: Similar loading is done on also other pages, so a generic component for doing that would make sense.

File: src/cljs/rems/roles.cljs (hunk 4 above)
Comment (luontola) [reply src/cljs/rems/roles.cljs]: At least they still take roles as parameters, so they are to some extent about roles. I'm feeling too lazy to change the namespace name. 😉

**Pull request merged**